### PR TITLE
Program List and Creation UI

### PR DESCRIPTION
**Pull request by hemant10yadav**
https://dimagi.atlassian.net/browse/CCCT-424

Added program app with list view, create and update view.

Screenshot 2024-07-30 at 08 17 03
Screenshot 2024-07-30 at 08 17 49
Screenshot 2024-07-30 at 08 27 47 1

### CODE REVIEW

File: commcare_connect/program/models.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+from commcare_connect.opportunity.models import DeliveryType
+from commcare_connect.utils.db import BaseModel, slugify_uniquely
+
+
+class Program(BaseModel):
+    name = models.CharField(max_length=255)
+    slug = models.SlugField(max_length=255, unique=True)
+    description = models.CharField()
+    delivery_type = models.ForeignKey(DeliveryType, null=True, blank=True, on_delete=models.DO_NOTHING)
```
Comment (calellowitz) [inline commcare_connect/program/models.py:11]: This should be required, (and I think should use `models.PROTECT`)

**Review by calellowitz — COMMENTED** [review]
couple small things but generally looks good

File: commcare_connect/program/models.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+from commcare_connect.opportunity.models import DeliveryType
+from commcare_connect.utils.db import BaseModel, slugify_uniquely
+
+
+class Program(BaseModel):
+    name = models.CharField(max_length=255)
+    slug = models.SlugField(max_length=255, unique=True)
+    description = models.CharField()
+    delivery_type = models.ForeignKey(DeliveryType, on_delete=models.PROTECT)
```
Comment (sravfeyn) [inline commcare_connect/program/models.py:11]: Don't we need `organization` on this?

**Review by sravfeyn — APPROVED** [review]
LGTM

File: commcare_connect/program/models.py (hunk 2 above)
Comment (hemant10yadav) [reply commcare_connect/program/models.py:11]: We do it is added in the Managed Opportunity Creation PR.

**Comment by hemant10yadav** [comment]
Test related to program are added in this PR
[PR-379](https://github.com/dimagi/commcare-connect/pull/379)

**Review by calellowitz — APPROVED** [review]
looks good

**Review by pxwxnvermx — APPROVED** [review]

**Pull request merged**